modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/dht/topology/PartitionsEvictManager.java

```diff
@@ -111,13 +103,10 @@
      * @param grp Group context.
      */
     public void onCacheGroupStopped(CacheGroupContext grp) {
-        GroupEvictionContext grpEvictionCtx = evictionGroupsMap.remove(grp.groupId());
+        GroupEvictionContext grpEvictionCtx =
+            evictionGroupsMap.computeIfAbsent(grp.groupId(), p -> new GroupEvictionContext(grp));
```

Comment:
Looks like we have a slow potential memory leak here: the value is removed from the map only on group start, so if I start and then stop unique caches, the map will grow.

Reply:
I'll create a separate ticket to resolve this.